sdk/spring/spring-cloud-azure-feature-management/README.md

```diff
@@ -192,10 +192,159 @@ feature-management:
+
+* Parameters
+
+  Property | Relevance | Description
+  -----------|-------|-----
+  **type** | Required  | Two valid values: `Daily`, `Weekly`.
+  **interval** | Optional  | Specifies the interval between each start time of occurrence. Default value is 1. <br/>For example, if the interval is 2 with `Daily` type, and current occurrence is 2:00 AM ~ 3:00 AM on 2024/05/11, then the next occurrence should be 2:00 AM ~ 3:00 AM on 2024/05/13
+  **daysOfWeek** | Optional  | Specifies on which day(s) of the week the event occurs. It's required when `Weekly` type, not applicable for `Daily` type. 
+  **firstDayOfWeek** | Optional  | Specifies which day is considered the first day of the week. Default value is `Sunday`. Not applicable for `Daily` type. 
+  
+  * Example
+    * Daily
+  
+        The daily recurrence pattern causes the time window to repeat based on interval between each occurrence.
+
+        The following example will repeat from 2:00 AM to 3:00 AM on every 2 days
+
+        ```yaml
+        start: "Mon, 13 May 2024 02:00:00 GMT",
+        end: "Mon, 13 May 2024 03:00:00 GMT",
+        recurrence:
+          pattern:
+            type: "Daily",
+            interval: 2
+          range:
+            type: "NoEnd"
+        ```
+
+    * Weekly
+  
+      The weekly recurrence pattern causes the time window to repeat on the same day or days of the week, based on the number of weeks between each set of occurrences.
+
+      The following example will repeat from 2:00 AM to 3:00 AM on every other Monday and Tuesday
+
+        ```yaml
+        start: "Mon, 13 May 2024 02:00:00 GMT",
+        end: "Mon, 13 May 2024 03:00:00 GMT",
+        recurrence:
+          pattern:
+            type: "Weekly",
+            interval: 2,
+            daysOfWeek: 
+              - Monday
+              - Tuesday
+          range:
+            type: "NoEnd"
+        ```
+
+    **Note:** `start` must be a valid first occurrence which fits the recurrence pattern. For example, if we define to repeat on every other Monday and Tuesday, then the start time should be in Monday or Tuesday. <br /> Additionally, the duration of the time window cannot be longer than how frequently it occurs. For example, it is invalid to have a 25-hour time window recur every day.
+
+#### Recurrence Range
+
+There are three possible recurrence range type: `NoEnd`, `EndDate` and `Numbered`.
+
+* Parameter
+  
+    Property | Relevance | Description                                                                           |
+    -----------|---------------|-------------
+    **type** | Required  | Three valid values: `NoEnd`, `EndDate` and `Numbered`.
+    **endDate** | Optional  |  Specifies the date time to stop applying the pattern. Note that as long as the start time of the last occurrence falls before the end date, the end time of that occurrence is allowed to extend beyond it. <br /> It's required for `EndDate` type, not applicable for `NoEnd` and `Numbered` type.
```

Comment:
endDate currently has a default value. It's not "required", The end date just has to be before `1970-01-25T20:31:23.647Z`

Reply:
I removed the default value of `endDate`. It sounds not reasonable that we have some default value.